agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/quickpulse/QuickPulseDataCollector.java

```diff
@@ -176,6 +176,11 @@ synchronized FinalCounters peek() {
   }
 
   public void add(TelemetryItem telemetryItem) {
+    if (telemetryClient == null) {
```

Comment:
I think this will just stop collecting data. But we also need to stop quick pulse pings?

Reply:
> we also need to stop quick pulse pings?

that part is already handled